src/parse-open-rpc-document.ts

```diff
@@ -63,75 +63,79 @@ const defaultParseOpenRPCDocumentOptions = {
   validate: true,
 };
 
-/**
- * Resolves an OpenRPC document from a variety of input types. The resolved OpenRPC document
- * will be dereferenced and validated against the [meta-schema](https://github.com/open-rpc/meta-schema).
- *
- * @param schema The OpenRPC document or a reference to one.
- *
- * If schema is an object, it will use this as the openrpc document literally.
- * If schema is a string, it may fall under 3 other categories:
- *   1. schema is an OpenRPC document as a json string.
- *   2. schema is a url that resolves to an OpenRPC document.
- *   3. schema is a file path, where the file at the path contains an OpenRPC document.
- *
- * @param options Parser options. See [[IParseOpenRPCDocumentOptions]]
- *
- * @returns The same OpenRPC Document that was passed in, but with all $ref's dereferenced.
- *
- * @throws [[OpenRPCDocumentValidationError]]
- * @throws [[OpenRPCDocumentDereferencingError]]
- *
- * @example
- * ```typescript
- *
- * const { OpenRPC } from "@open-rpc/meta-schema"
- * const { parseOpenRPCDocument } from "@open-rpc/schema-utils-js";
- *
- * try {
- *   const fromUrl = await parseOpenRPCDocument("example.com/openrpc.json") as OpenRPC;
- *   const fromFile = await parseOpenRPCDocument("example.com/openrpc.json") as OpenRPC;
- *   const fromString = await parseOpenRPCDocument('{ "openrpc": "1.0.0", ... }') as OpenRPC;
- *   const fromCwd = await parseOpenRPCDocument() as types.OpenRPC; // default
- * } catch (e) {
- *   // handle validation errors
- * }
- * ```
- *
- */
-export default async function parseOpenRPCDocument(
-  schema: string | OpenRPC = "./openrpc.json",
-  options: IParseOpenRPCDocumentOptions = defaultParseOpenRPCDocumentOptions,
-): Promise<OpenRPC> {
-  let parsedSchema: OpenRPC;
+const makeParseOpenRPCDocument = (fetchUrlSchema: TGetOpenRPCDocument, readSchemaFromFile: TGetOpenRPCDocument) => {
+  /**
```

Comment:
it would read a lot cleaner if you typed the interface and put the typedoc on it, instead of here.

```
/**
 * <TYPEDOC GOES HERE>
 */
export type ParseOpenRPCDocument = (
  options: IParseOpenRPCDocumentOptions = defaultParseOpenRPCDocumentOptions,
  schema: string | OpenRPC = "./openrpc.json",
) => Promise<OpenRPC>;

makeParseOpenRPCDocument = (
  fetchUrlSchema: TGetOpenRPCDocument, 
  readSchemaFromFile: TGetOpenRPCDocument
): ParseOpenRPCDocument => { ... }

```